### ISSUE TITLE
2.11.0: Spritesheets with only 2 frames show whole sheet

### ISSUE THREAD

**Issue by FostUK** [photonstorm/phaser-ce#559]
Moving from v2.10.5 to 2.11.0 - has caused buttons made from a 2 frame spritesheet to stop working.

**2.10.5:**
![2 10 5](https://user-images.githubusercontent.com/961406/42324414-0643d228-805b-11e8-8104-ee05385e5242.jpg)

**2.11.0**
![2 11 0](https://user-images.githubusercontent.com/961406/42324448-21090dda-805b-11e8-9b15-16b74afe3cf2.jpg)

**Example Button sprite sheet (180X180 with 4 pixel spacing)**
![play](https://user-images.githubusercontent.com/961406/42324681-b6e20bfe-805b-11e8-8b63-e605a6a44a32.png)

I think I've narrowed it down to this commit: https://github.com/photonstorm/phaser-ce/commit/4489fdd4e7235f58b25c281faec261064226f3b7#diff-52abcff05316a7bec5cea054ab01bd4d

With a 2 frame animation the row or column is zero so frame total ends up zero.

I fixed it by changing lines 89 and 80 of _AnimationParser.js_ to:

```
        var row = Math.floor((width - margin) / (frameWidth + spacing)) + 1;
        var column = Math.floor((height - margin) / (frameHeight + spacing)) + 1;
```
Which adds 1 to each. Although I'm unsure if this is a workable solution.

**Comment by samme** [comment]
What are you using for `load.spritesheet(…)`?

**Comment by samme** [comment]
I tried that image in v2.10.5 and got the same result (but a different console warning).

It does work for me using the attached image and `spacing=4`. 

![frame size 180x180, spacing 4](https://user-images.githubusercontent.com/435681/42357710-17fae744-808e-11e8-8055-6f626101d737.png)

**Comment by FostUK** [comment]
Just checked and 2.10.5 is definitely still works for us.
I checked adding 4 pixels spacing to the right of the image in 2.11.0 but it doesn't change anything.

We're not loading the spritesheet directly though, we are using `game.load.pack` to load an asset pack so perhaps it's this path where something is different. 
Button json looks like this:

```
        {
            "type": "spritesheet",
            "key": "play",
            "url": "gel/desktop/play.png",
            "frameWidth": 180,
            "frameHeight": 180,
            "frameMax": 2,
            "margin": 0,
            "spacing": 4
        },
```
I'll try and debug it a bit more but any suggestions where to look welcome :)

**Comment by samme** [comment]
The `row` and `col` calculations should be identical in v2.10.5 and v2.11.0, so I'm not sure how the same image could have worked before. 

Do you see a console warning in either version?

**Comment by samme** [comment]
I found it: the difference is that before v2.11.0, `frameMax` would force the last frame event if it was too small.

I'm going to leave it how it is, because it's more consistent this way, but I've added some new warnings.

If you use `spacing`, every frame in the image must have spacing on its right and bottom edges, including last-in-row and last-in-column frames.

![Sample spritesheet with margin=2, spacing=4](https://user-images.githubusercontent.com/435681/42422007-acb3d358-8293-11e8-921c-59bf85b28ee5.png)
